<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns="http://www.tibco.com/CommonLE2/namespace/private/Common" xmlns:pfx0="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Exception/ExceptionResolution" xmlns:pfx1="http://www.tibco.com/CommonLE2/Public/Common/Dummy" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:wsdlPfx="http://www.tibco.com/CommonLE2/namespace/ExceptionManagement/GUIServices" xmlns:ns12="http://www.tibco.com/CommonLE/namespace/Common" xmlns:ns5="http://www.tibco.com/CommonLE2/namespace/Private/Exception" xmlns:pfx="http://www.tibco.com/pe/DeployedVarsType" xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns2="http://www.tibco.com/CommonLE2/namespace/Administration/GUIServices/Applications" xmlns:ns1="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:ns4="http://www.tibco.com/pe/EngineTypes" xmlns:ns3="http://www.tibco.com/CommonLE2/Public/Common">
    <wsdl:import namespace="http://www.tibco.com/CommonLE2/namespace/ExceptionManagement/GUIServices" location="/SharedResources/WSDL/GUI/ExceptionManagement/intfResolveExceptions.wsdl"/>
    <pd:name>ExceptionManagement/GUIServices/wsResolveExceptions.process</pd:name>
    <pd:startName>Start</pd:startName>
    <pd:startType wsMsgRef="wsdlPfx:ExceptionResolutionRecordList"/>
    <pd:startX>151</pd:startX>
    <pd:startY>49</pd:startY>
    <pd:returnBindings>
        <wsdlPfx:Dummy/>
    </pd:returnBindings>
    <pd:endName>End</pd:endName>
    <pd:endType wsMsgRef="wsdlPfx:Dummy"/>
    <pd:endX>449</pd:endX>
    <pd:endY>49</pd:endY>
    <pd:errorSchemas>
        <Schema0 wsMsgRef="wsdlPfx:Error"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1284597943964</pd:targetNamespace>
    <pd:activity name="Call Process">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>275</pd:x>
        <pd:y>50</pd:y>
        <config>
            <processName>/ExceptionManagement/DataAccess/ResolveExceptions.process</processName>
            <processNameXPath/>
        </config>
        <pd:inputBindings>
            <ns5:ExceptionResolutionRecords>
                <xsl:for-each select="$Start/wsdlPfx:ExceptionResolutionRecordList/parameters/pfx0:ExceptionResolutionRecordList/pfx0:ExceptionResolutionRecord">
                    <ns5:ExceptionResolution>
                        <ns5:ExceptionID>
                            <xsl:value-of select="pfx0:ExceptionID"/>
                        </ns5:ExceptionID>
                        <ns5:Resolution>
                            <xsl:value-of select="pfx0:ResolutionDesc"/>
                        </ns5:Resolution>
                        <ns5:Status>
                            <xsl:value-of select="pfx0:Status"/>
                        </ns5:Status>
                        <ns5:TransactionUpdate>
                            <xsl:value-of select="pfx0:TransactionUpdate"/>
                        </ns5:TransactionUpdate>
                    </ns5:ExceptionResolution>
                </xsl:for-each>
            </ns5:ExceptionResolutionRecords>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch All Errors">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>159</pd:x>
        <pd:y>318</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <catchAll>true</catchAll>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Generate Internal Error">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>487</pd:x>
        <pd:y>326</pd:y>
        <config>
            <faultName>Schema0</faultName>
        </config>
        <pd:inputBindings>
            <ns1:ActivityInput>
                <message>
                    <xsl:value-of select="$_error/ns4:ErrorReport/Msg"/>
                </message>
                <messageCode>
                    <xsl:value-of select="$_error/ns4:ErrorReport/MsgCode"/>
                </messageCode>
                <data>
                    <wsdlPfx:Error>
                        <parameters>
                            <ns3:Error>
                                <ns3:Message>
                                    <xsl:value-of select="$_error/ns4:ErrorReport/Msg"/>
                                </ns3:Message>
                                <ns3:Code>
                                    <xsl:value-of select="$_error/ns4:ErrorReport/MsgCode"/>
                                </ns3:Code>
                            </ns3:Error>
                        </parameters>
                    </wsdlPfx:Error>
                </data>
            </ns1:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Local Exception Handler">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>324</pd:x>
        <pd:y>325</pd:y>
        <config>
            <processName>/Common/Local Exception Handler.process</processName>
        </config>
        <pd:inputBindings>
            <ns12:Exception>
                <ErrorMessage>
                    <xsl:value-of select="$_error/ns4:ErrorReport/Msg"/>
                </ErrorMessage>
                <ErrorCode>
                    <xsl:value-of select="$_error/ns4:ErrorReport/MsgCode"/>
                </ErrorCode>
                <ns12:Error>
                    <xsl:for-each xmlns:xsl="http://www.w3.org/1999/XSL/Transform" select="$_error/ns4:ErrorReport">
                        <ns12:ErrorReport>
                            <StackTrace>
                                <xsl:value-of select="StackTrace"/>
                            </StackTrace>
                            <Msg>
                                <xsl:value-of select="Msg"/>
                            </Msg>
                            <FullClass>
                                <xsl:value-of select="FullClass"/>
                            </FullClass>
                            <Class>
                                <xsl:value-of select="Class"/>
                            </Class>
                            <ProcessStack>
                                <xsl:value-of select="ProcessStack"/>
                            </ProcessStack>
                            <MsgCode>
                                <xsl:value-of select="MsgCode"/>
                            </MsgCode>
                        </ns12:ErrorReport>
                    </xsl:for-each>
                </ns12:Error>
            </ns12:Exception>
        </pd:inputBindings>
    </pd:activity>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Call Process</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Call Process</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch All Errors</pd:from>
        <pd:to>Local Exception Handler</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Local Exception Handler</pd:from>
        <pd:to>Generate Internal Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>